gui/default/syncthing/folder/ignoreTreeService.js

```diff
@@ -0,0 +1,100 @@
+angular.module('syncthing.folder')
+    .service('IgnoreTree', function (
+        Ignores,
+        Browse,
+    ) {
+        'use strict';
+
+        // Bind methods directly to the controller so we can use controllerAs in template
+        var self = this;
+
+        // public definitions
+        self.tree = null;
+
+        self.refresh = function(folderId) {
+            var promise = Browse.refresh(folderId).then(function(response) {
+                return response.files.map(buildNode);
+            });
+
+            if (self.tree) {
+                self.tree.reload(promise);
+            } else {
+                self.tree = $("#ignore-tree").fancytree({
+                    extensions: ["table"],
+                    checkbox: true,
+                    selectMode: 2,
+                    table: {
+                        indentation: 20,
+                        nodeColumnIdx: 1,
+                        checkboxColumnIdx: 0,
+                    },
+                    debugLevel: 2,
+                    source: promise,
+                    lazyLoad: function (event, data) {
+                        var prefix = data.node.data.file.path;
+                        data.result = Browse.refresh(folderId, prefix).then(function(response) {
+                            return response.files.map(buildNode);
+                        });
+                    },
+                    select: function (event, data) {
+                        toggle(data.node);
+                    },
+                }).fancytree("getTree");
+            }
+        }
+
+        self.update = function() {
+            if (self.tree) updateNodes(self.tree.rootNode.children)
+        };
+
+        function buildNode(file) {
+            var match = Ignores.matchingPattern(file);
+
+            return {
+                // Fancytree keys
+                title: file.name,
+                selected: (!match || match.isNegated),
+                key: file.path,
+                lazy: !file.isFile,
+                folder: !file.isFile,
+                // Data keys
+                file: file,
```

Comment:
I think file should be path here? As we're calling buildNode for directories too?
But then you have path.path, so perhaps this should be entry or something like that.

Reply:
Yes, the issue is about the name of the variable.

Reply:
Got it - there are a number of places file is referred to this way, I'll go with entry instead